environments/openstack/README.md

```diff
@@ -1,17 +1,135 @@
-# Provisioning with Openstack
+# Deployment on OpenStack
 
-## How to run it ?
+In this tutorial you will learn how to set up MedTagger on OpenStack
+using Terraform and Ansible. 
 
-Download [`Openstack RC file`](https://docs.openstack.org/zh_CN/user-guide/common/cli-set-environment-variables-using-openstack-rc.html) from your Openstack instance. It holds all needed environmental variables.
+### Generating SSH keys
+
+Proceed to your OpenStack Instance and go to **Computations - Access and Security - Key pairs** tab.
+
+Click on **Create Key Pair** button and create 2 sets of key pairs.
+
+**Important**: By default our configurations scripts assume that the keys are named `app-key`
+and `db-key`. If you want to use different names, please change them in `main.tf` and 
+`group_vars/all.yml` files.
+
+The download of the private keys should start automatically after creating them. If not,
+download private keys manually and place then in **Ansible directory.**
+
+Remember to change file mode:
 
-Source the file:
+```bash
+$ chmod 0600 db-key.pem
+```
+
+### Machines provisioning
+
+After this step you will have:
+- 2 (Ubuntu 18) instances (AppHost and DbHost) available on OpenStack,
+- AppHost will have externally available IP address,
+- Network connectivity between your App and Db host,
+- 4 volumes set up (App, PostgreSQL, Db and Cassandra),
+- Security Groups enabling certain ports for different services.
+
+**Important**: Head to the `main.tf` file to see which of them are configurable. Keep in mind that
```

Comment:
Could it be somehow parametrized? Maybe set through environment settings somehow? Is changing files (that are tracked by GIT) the only and proper way to do so?

Reply:
I have a feeling that you are mentioning some other file...

The `main.tf` hold variables such as `image_id` or `external_network_name` that are unique for each of the Openstack instances.

Could you elaborate then what you mean exactly ? Because I'm suspecting you are talking about modyfying .backend or docker-compose.yaml.

Reply:
As stated offline, I will create the `.example.main.tf` file and explain in the README how to fill it up. I will also add `main.tf` to the gitignore, to make sure that no one will accidentally push his/her private Openstack config.